### ISSUE TITLE
Firestore does not support FB Messenger integration

### ISSUE THREAD

**Issue by Mistic92** [jovotech/jovo-framework#776]
## I'm submitting a...

- [X] Bug report
- [ ] Feature request
- [ ] Documentation issue or request
- [ ] Other... Please describe:

## Expected Behavior
Use firestore integration with FB Messenger

## Current Behavior
I'm getting error
```
Error -----------------------------------------------------------------
  Message:
  Argument "data" is not a valid Document. Couldn't serialize object of type "TextQuickReply" (found in field userData.context.prev.`0`.response.output.FacebookMessenger.Overwrite.QuickReplies.`0`). Firestore doesn't support JavaScript objects with custom prototypes (i.e. objects that were created via the "new" operator).
  Stack:
  Error: Argument "data" is not a valid Document. Couldn't serialize object of type "TextQuickReply" (found in field userData.context.prev.`0`.response.output.FacebookMessenger.Overwrite.QuickReplies.`0`). Firestore doesn't support JavaScript objects with custom prototypes (i.e. objects that were created via the "new" operator).
      at Validator.<computed> (/root/projects/xxxxxxxxxx/node_modules/@google-cloud/firestore/build/src/validate.js:92:27)
      at WriteBatch.set (/root/projects/xxxxxxxxxx/node_modules/@google-cloud/firestore/build/src/write-batch.js:225:25)
      at DocumentReference.set (/root/projects/xxxxxxxxxx/node_modules/@google-cloud/firestore/build/src/reference.js:340:27)
      at Firestore.save (/root/projects/xxxxxxxxxx/node_modules/jovo-db-firestore/src/Firestore.ts:163:18)
      at JovoUser.saveDb (/root/projects/xxxxxxxxxx/node_modules/jovo-framework/src/middleware/user/JovoUser.ts:469:35)
      at Middleware.run (/root/projects/xxxxxxxxxx/node_modules/jovo-core/src/core/Middleware.ts:85:20)
      at App.handle (/root/projects/xxxxxxxxxx/node_modules/jovo-core/src/core/BaseApp.ts:277:43)
      at processTicksAndRejections (internal/process/task_queues.js:97:5)
      at async Promise.all (index 0)
      at App.jovo_core_1.BaseApp.handle (/root/projects/xxxxxxxxxx/node_modules/jovo-platform-facebookmessenger/src/FacebookMessenger.ts:346:9)
   ----------------------------------------------------------------------
```

My "user" config looks like this
```js
  user: {
    updatedAt: true,
    dataCaching: false,
    implicitSave: true,
    metaData: {
      enabled: true,
      lastUsedAt: true,
      sessionsCount: true,
      createdAt: true,
      requestHistorySize: 4,
      devices: true,
    },
    sessionData: {
      enabled: false,
      data: false,
      id: true,
    },
    context: {
      enabled: true,
      prev: {
        size: 3,
        request: {
          intent: true,
          state: true,
          inputs: true,
          timestamp: true,
        },
        response: {
          speech: true,
          reprompt: true,
          state: true,
          output: true,
        },
      },
    },
  },
```

### Error log
```
If you have an error log, please paste it here.
```

## Your Environment

* Jovo Framework version used: 1.x.x
* Operating System:
Docker/Win10
```js
    "jovo-analytics-googleanalytics": "^3.0.22",
    "jovo-core": "^3.0.17",
    "jovo-db-filedb": "^3.0.18",
    "jovo-db-firestore": "^3.0.18",
    "jovo-framework": "3.0.19",
    "jovo-nlu-dialogflow": "^3.0.17",
    "jovo-platform-facebookmessenger": "^3.0.22",
    "jovo-plugin-debugger": "^3.0.17",
```

**Comment by aswetlow** [comment]
Sorry, took a little longer.
Could you provide the relevant part of your user data object?


```javascript
app.hook('before.user.save', async (error, host, jovo) => {
    console.log(JSON.stringify(jovo.$output, null, '\t'));
});
```

**Comment by m-ripper** [comment]
Hey, I could check on the issue today and found that this error was caused by an underlying issue in `JovoUser`.

The error is caused by `JovoUser` trying to save a session-id with the value `undefined`, which seems to cause more severe issues for some database-integrations. 

I have opened a PR that fixes this issue #818.

Until the changes are merged and published, you can avoid this issue by setting `user.sessionData.id` to `false`.

Let me know if this does not fix your issue.

**Comment by aswetlow** [comment]
Closing due to inactivity. Please re-open if it still persists.
